### ISSUE TITLE
Email notification Send Test sometimes generate an error because of fromName auto-generated content

### ISSUE THREAD

**Issue by romainpoirier** [verbb/formie#523]
**Description**
The _Email Preview From_ is auto-generated with _Lorem Ipsum_. Each time I click the _Refresh_ button, I can see a new _From_ in the preview below. This issue is that this _From_ value is usually very long, which cause an _Error sending test email_. on _Send Test Email_ (even if the email is finally sent).

This is the kind of error:
```
Notification email “Test” could not be sent for submission “new”. Error: SQLSTATE[22001]: String data, right truncated: 1406 Data too long for column 'fromName' at row 1 The SQL being executed was: INSERT INTO `craft_formie_sentnotifications` (`id`, `title`, `formId`, `submissionId`, `notificationId`, `subject`, `to`, `cc`, `bcc`, `replyTo`, `replyToName`, `from`, `fromName`, `body`, `htmlBody`, `info`, `dateCreated`, `dateUpdated`, `uid`) VALUES (284, 'Test', 54, NULL, 3, 'This is a test', 'romain@test.eu', NULL, NULL, '', '', 'braun.leda@test.biz', 'Molestias inventore debitis aliquam. Architecto rerum laboriosam nam ratione et iste. Optio vitae temporibus dicta perferendis.Expedita adipisci et et adipisci. Ipsam minus impedit aliquam consequatur a vero reprehenderit. Quia voluptatem dolorum velit iure.', 'The body content for this notification.\n', '
The body content for this notification.

', '{\"formieVersion\":\"Formie dev-craft-3\",\"craftVersion\":\"Craft Pro 3.7.0\",\"ipAddress\":\"::1\",\"userAgent\":\"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.114 Safari/537.36\",\"transportType\":\"Mandrill\"}', '2021-07-14 12:47:02', '2021-07-14 12:47:02', 'd556f9cb-280b-4f2e-b7c7-9652a6abf544') /vendor/yiisoft/yii2/db/Schema.php:678
```

**Steps to reproduce**
1. Setup an Email Notification
2. Send Test Email

### Form settings
- Multi-page form: Yes
- Submission Method: Page Reload
- Client-side Validation: Yes
- Custom Form Templates: No

**Additional info**
- Plugin version: dev-craft-3 as 1.4.9
- Craft version: 3.7.0
- Multi-site: No

**Comment by engram-design** [comment]
You must be mapping a field in your form to the "From" field right? By default it uses the site name (if you don't select anything to map to it). If you map it to a Name field, it should generate a name, but it looks like you might be mapping it to a Single-Line Text field, which might be a bit risky. Mapping it to a Name field produces expected results.

If that's the case, we'll need to override what you're mapping it to, to force a name to be generated, rather than rely on the generated mapped content

**Comment by romainpoirier** [comment]
You're right, I am mapping a _First Name_ + _Last Name_ which are set using `Single-line Text` fields.
I missed the utility of `Name` fields. I didn't thought that it could break something in the email notifications.

**Comment by engram-design** [comment]
It shouldn't break the actual notification, only just the test/preview functionality where we generate fake content.

**Comment by romainpoirier** [comment]
FYI I still get an error even if the fields in the _Email Notification_ > _Advanced_ are not changed (so they are all blank).
Despite this error, the **Send Test Email** is still working, and the email is sent.

![](https://i.imgur.com/591DrI5.png)
![](https://i.imgur.com/9eeEv5m.png)

```
Notification email “Test” could not be sent for submission “new”. Error: SQLSTATE[HY000]: General error: 1366 Incorrect string value: '\xF0\x9F\x9A\xA7 C...' for column 'fromName' at row 1 The SQL being executed was: INSERT INTO `craft_formie_sentnotifications` (`id`, `title`, `formId`, `submissionId`, `notificationId`, `subject`, `to`, `cc`, `bcc`, `replyTo`, `replyToName`, `from`, `fromName`, `body`, `htmlBody`, `info`, `dateCreated`, `dateUpdated`, `uid`) VALUES (294, 'Test', 289, NULL, 5, 'Test – Notification', 'romain@test.com', NULL, NULL, '', '', 'hello@test.com', 'My Website', 'EMAIL\nkbeatty@sample.com\n\nSINGLE-LINE TEXT\nEt culpa omnis enim aut qui quidem aperiam. Sit dicta ex et. Voluptas\nquis repellat et quo nihil et magnam. Est nisi eum repellat illum sed\nmolestias quibusdam. Ut esse ut et qui et eos amet.\n', '
Email
kbeatty@sample.com

Single-line Text
Et culpa omnis enim aut qui quidem aperiam. Sit dicta ex et. Voluptas quis repellat et quo nihil et magnam. Est nisi eum repellat illum sed molestias quibusdam. Ut esse ut et qui et eos amet.

', '{\"formieVersion\":\"Formie dev-craft-3\",\"craftVersion\":\"Craft Pro 3.7.0\",\"ipAddress\":\"::1\",\"userAgent\":\"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.114 Safari/537.36\",\"transportType\":\"Mandrill\"}', '2021-07-14 15:31:19', '2021-07-14 15:31:19', 'e8ee4e4f-4bb2-45ff-aff3-f087b107cd1e') /vendor/yiisoft/yii2/db/Schema.php:678
```

**Comment by engram-design** [comment]
Well that doesn't make sense, that very error shows it's trying to insert `My Website` as the value for `fromName` which is perfectly fine. I can't see where `\xF0\x9F\x9A\xA7 C...` is coming from, which frustratingly is truncated so it's not helpful...

You mention that you're getting this error but the **Send Test Email** function works. Is this error just showing up when you view "Preview" (or hit "Refresh"), but sending a test is fine? It looks like it's all related to the sent notifications functionality..

**Comment by engram-design** [comment]
Should be fixed for the next release. To get the fix early, change your `verbb/formie` requirement in `composer.json` to:

```
"require": {
  "verbb/formie": "dev-craft-3 as 1.4.9",
  "...": "..."
}
```

Then run `composer update`.

**Comment by engram-design** [comment]
Fixed in [1.4.10](https://github.com/verbb/formie/releases/tag/1.4.10)

**Comment by romainpoirier** [comment]
I still get the error with `1.4.10`.

Yes, sending the test is fine, but this kind of error is still showing up under the _Send Test Email_ fieldset:

```
Notification email “Test” could not be sent for submission “new”. Error: SQLSTATE[HY000]: General error: 1366 Incorrect string value: '\xF0\x9F\x9A\xA7 C...' for column 'fromName' at row 1 The SQL being executed was: INSERT INTO `craft_formie_sentnotifications` (`id`, `title`, `formId`, `submissionId`, `notificationId`, `subject`, `to`, `cc`, `bcc`, `replyTo`, `replyToName`, `from`, `fromName`, `body`, `htmlBody`, `info`, `dateCreated`, `dateUpdated`, `uid`) VALUES (309, 'Test', 289, NULL, 5, 'Test – Notification', 'romain@example.com', NULL, NULL, '', '', 'hello@example.com', '🚧 Test', 'EMAIL\ncontact@example.com\n\nSINGLE-LINE TEXT\nUt suscipit nam harum ut ea sed. Aut quidem natus quisquam possimus\nblanditiis a ducimus porro. Aut molestiae possimus laboriosam ea.\nQuidem ut aliquid qui quo vitae.\n', '
Email
moore.una@hotmail.com

Single-line Text
Ut suscipit nam harum ut ea sed. Aut quidem natus quisquam possimus blanditiis a ducimus porro. Aut molestiae possimus laboriosam ea. Quidem ut aliquid qui quo vitae.

', '{\"formieVersion\":\"Formie 1.4.10\",\"craftVersion\":\"Craft Pro 3.7.1\",\"ipAddress\":\"::1\",\"userAgent\":\"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.114 Safari/537.36\",\"transportType\":\"Mandrill\"}', '2021-07-15 08:55:30', '2021-07-15 08:55:30', '24dd9a99-3cf1-4d88-b462-fa143435e305') /vendor/yiisoft/yii2/db/Schema.php:678
```

**Comment by engram-design** [comment]
Hmmm, I can't for the life of me figure out why it's setting the string to `\xF0\x9F\x9A\xA7 C...`.

Would this be your site name? And if so, any special characters in there? From what I can tell, this is an emoji for the construction sign. Likely added by SEOmatic?

**Comment by engram-design** [comment]
Should be fixed for the next release. To get the fix early, change your `verbb/formie` requirement in `composer.json` to:

```
"require": {
  "verbb/formie": "dev-craft-3 as 1.4.10",
  "...": "..."
}
```

Then run `composer update`.

**Comment by romainpoirier** [comment]
Yes, there is an emoji (🚧) in the site name for local dev. It is not added by SEOmatic, just setup by myself in the env file (`SENDER_NAME` constant). Unfortunately, after updating to `dev-craft-3 as 1.4.10` the problem is still there.

**Comment by engram-design** [comment]
Ah, I see - that's what I ended up testing with anyway. But good point, my fix only handled it **before** parsing an environment variable, not after. A `composer update` should fix that.

**Comment by romainpoirier** [comment]
It is now fixed using `dev-craft-3 as 1.4.10`. Thank you 👏  !

**Comment by engram-design** [comment]
Fixed in [1.4.11](https://github.com/verbb/formie/releases/tag/1.4.11)
